common/src/main/java/com/LubieKakao1212/opencu/common/device/TrackerBase.java

```diff
@@ -72,6 +78,35 @@ public void tick(BlockEntityModularFrame frame, IDeviceState state, World world,
                     if (filters.contains(entity.getName().getString())) {
                         continue;
                     }
+                    if (entity.isPlayer()) {
+                        PlayerEntity player = (PlayerEntity) entity;
+                        if (player.isCreative() || player.isSpectator()) {
+                            continue;
+                        }
+                    }
+                    if (entity.isDead()){
+                        continue;
+                    }
+
+                    // Check if there's a block obstructing the view by reversing the raycast direction
+                    Vec3d start = entity.getBoundingBox().getCenter();
+                    Vec3d end = pos.toCenterPos();
+                    BlockHitResult blockHitResult = world.raycast(new RaycastContext(
+                            start,
+                            end,
+                            RaycastContext.ShapeType.COLLIDER,
+                            RaycastContext.FluidHandling.NONE,
+                            entity
+                    ));
+
+                    //Rather silly but it works
+                    if (blockHitResult.getType() == HitResult.Type.BLOCK) {
+                        BlockPos hitPos = blockHitResult.getBlockPos();
+                        var blockState = world.getBlockState(hitPos);
+                        if (!blockState.getBlock().toString().contains("modular_frame")){
```

Comment:
Checks if the hit block is not a Modular Frame i guess?

Reply:
...

Reply:
because the raycast will hit the block itself, it's supposed to be used for entity to entity cast but since it starts at the center it hits itself. So I cast from entity to the block and if it doesnt hit anything other than the block itself it considers it a valid target.

Reply:
this check is incorrect for several reasons:
- It checks for any "modular_frame" and not the current one
- It checks if the BlockState contains "modular_frame" and not if it is an actual modular_frame block so it would trigger even for things from other mods which have "modular_frame" in their name or even have a state property (key or value) containing "modular_frame"

You could instead check if hit position is equal to current position which is the simplest option, I added a suggestion for it

Reply:
Hey it was 2am and I was throwing shit at the wall haha, but totally agreed!

Reply:
You forgot to apply the suggestion.